legate/core/operation.py

```diff
@@ -410,21 +410,25 @@ def add_scalar_arg(
             ):
                 raise ValueError(f"{value} is not a valid scalar")
         elif isinstance(dtype, tuple):
-            assert len(dtype) == 1 and isinstance(dtype[0], ty.Dtype)
+            if not (len(dtype) == 1 and isinstance(dtype[0], ty.Dtype)):
+                raise TypeError(f"Unsupported type: {dtype}")
             if not isinstance(value, (tuple, Shape)):
                 raise ValueError(f"{value} is not a valid scalar")
             sanitized = ty.array_type(dtype[0], len(value))
         else:
             raise TypeError(f"Unsupported type: {dtype}")
 
-        if not (
-            sanitized.is_primitive
-            or sanitized == ty.string
-            or (
-                isinstance(sanitized, ty.FixedArrayDtype)
-                and sanitized.element_type.is_primitive
+        def is_supported_scalar_arg_type(dtype: ty.Dtype) -> bool:
+            return (
+                dtype.is_primitive
+                or dtype == ty.string
+                or (
+                    isinstance(dtype, ty.FixedArrayDtype)
+                    and dtype.element_type.is_primitive
+                )
```

Comment:
Suggest pulling this outside to the module level, so that there isn't the cost of defining this function added to every call to `add_scalar_arg`. As a side bonus the formatting will probably also improve with the extra room

Reply:
https://github.com/nv-legate/legate.core/pull/785/commits/119261a765d2ae3585b4053fa5ee12c9ef6c3f28